### ISSUE TITLE
Docs/examples for TF-Slim deployment mising

### ISSUE THREAD

**Issue by ogail** [tensorflow/models#1028]
## Please let us know which model this issue is about (slim/deployment/model_deploy.py)

The tutorial for multi-GPU for CIFAR-10 explains well the parameters with examples how they are used. However, model_deploy.py does not clarify what are the differences between `num_clones` and `num_replicas`. This is what I understand right now:

- use `num_clones` to specify how many GPU towers used to train the model
- use `num_replicas` to specify how many worker machines are used for training.

In this case how/where to provide the worker machine name/IP so the ps service can communicate with it?

There are couple of questions about that topic, which has not been answered:

- https://github.com/tensorflow/models/issues/403
- http://stackoverflow.com/questions/37894438/multiple-gpu-training-with-tensorfow-slim-learning
- https://github.com/tensorflow/models/issues/658

**Comment by aselle** [comment]
@wolffg,  could you look into finding someone to improve this tutorial?

**Comment by ogail** [comment]
Thanks also, I don't quite understand how the `num_readers` and `num_preprocessing_threads` in `slim/train_image_classifier.py` are used and how to set them. For example if there are 2 clones and 2 replicates should the `num_readers` be 8 and `num_preprocessing_threads` be 8 as well to assure replicas are busy? Are these numbers accurate? How much increasing and lowering these numbers would actually make a difference?

**Comment by gnavihs** [comment]
+1

**Comment by wolffg** [comment]
Assigning this to sguada, as he has the TODO(sguada) about this in the related file.

**Comment by MrWanter** [comment]
+1

**Comment by PangHua** [comment]
Found the answer.  
A: Fine-tuning batch normalization requires large batch size, and thus in the train.py we suggest setting `num_clones` (number of GPUs on one machine) and `train_batch_size` to be as large as possible.

https://github.com/tensorflow/models/blob/05ccaf88f7bd617a614f96a2414602f571cc2768/research/deeplab/g3doc/faq.md
